Modules/vc-module-inventory/VirtoCommerce.InventoryModule.Web/ExportImport/InventoryExportImport.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using VirtoCommerce.Domain.Inventory.Model.Search;
+using VirtoCommerce.InventoryModule.Core.Model;
+using VirtoCommerce.InventoryModule.Core.Model.Search;
+using VirtoCommerce.InventoryModule.Core.Services;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.ExportImport;
+using VirtoCommerce.Platform.Core.Settings;
+
+namespace VirtoCommerce.InventoryModule.Web.ExportImport
+{
+    public sealed class BackupObject
+    {
+        public BackupObject()
+        {
+            InventoryInfos = Array.Empty<InventoryInfo>();
+            FulfillmentCenters = Array.Empty<FulfillmentCenter>();
+        }
+        public InventoryInfo[] InventoryInfos { get; set; }
+        public FulfillmentCenter[] FulfillmentCenters { get; set; }
+    }
+
+    public sealed class InventoryExportImport
+    {
+        private readonly IInventoryService _inventoryService;
+        private readonly IInventorySearchService _inventorySearchService;
+        private readonly IFulfillmentCenterSearchService _fulfillmentCenterSearchService;
+        private readonly IFulfillmentCenterService _fulfillmentCenterService;
+        private readonly ISettingsManager _settingsManager;
+
+        private int? _batchSize;
+
+        private int BatchSize
+        {
+            get
+            {
+                if (_batchSize == null)
+                {
+                    _batchSize = _settingsManager.GetValue("Inventory.ExportImport.PageSize", 50);
+                }
+
+                return (int) _batchSize;
+            }
+        }
+
+        public InventoryExportImport(
+            IInventoryService inventoryService,
+            IFulfillmentCenterSearchService fulfillmentCenterSearchService,
+            IInventorySearchService inventorySearchService,
+            IFulfillmentCenterService fulfillmentCenterService,
+            ISettingsManager settingsManager
+            )
+        {
+            _inventoryService = inventoryService;
+            _fulfillmentCenterSearchService = fulfillmentCenterSearchService;
+            _fulfillmentCenterService = fulfillmentCenterService;
+            _inventorySearchService = inventorySearchService;
+            _settingsManager = settingsManager;
+        }
+
+        public async Task DoExport(Stream backupStream, Action<ExportImportProgressInfo> progressCallback)
```

Comment:
Export/Import should be switched to streamed version see the example
https://github.com/VirtoCommerce/vc-module-customer/blob/master/VirtoCommerce.CustomerModule.Web/ExportImport/CustomerExportImport.cs